src/controller/proxy/local.go

```diff
@@ -0,0 +1,182 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/manifestlist"
+	comHttpAuth "github.com/goharbor/harbor/src/common/http/modifier/auth"
+	"github.com/goharbor/harbor/src/controller/artifact"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/registry"
+	"io"
+	"time"
+)
+
+// localInterface defines operations related to local repo under proxy mode
+type localInterface interface {
```

Comment:
Why Pushblob, PushManifest and DeleteManifest does not have ctx as the parameter?

Reply:
Usually the api called outside the pkg need to have a ctx, for internal method and it doesn't use the ctx, no need to add it.